buildSrc/src/main/java/net/neoforged/neodev/utils/structure/ClassStructureVisitor.java

```diff
@@ -0,0 +1,80 @@
+package net.neoforged.neodev.utils.structure;
+
+import org.apache.commons.lang3.mutable.MutableInt;
+import org.objectweb.asm.ClassReader;
+import org.objectweb.asm.ClassVisitor;
+import org.objectweb.asm.FieldVisitor;
+import org.objectweb.asm.MethodVisitor;
+import org.objectweb.asm.Opcodes;
+import org.objectweb.asm.Type;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Map;
+import java.util.zip.ZipFile;
+
+public final class ClassStructureVisitor extends ClassVisitor {
+    private final Map<String, ClassInfo> classes;
```

Comment:
What I had in mind was constructing the map here instead of passing a reference to the constructor of the visitor to make the data flow somewhat more obvious. As you prefer.

In that case you can even make the constructor private and just expose a static method that takes a File and returns the map.